src/PowerShellTips/2023-09-22-avoid-array-addition.ps1

```diff
@@ -0,0 +1,31 @@
+$tip = [tiPS.PowerShellTip]::new()
+$tip.CreatedDate = [DateTime]::Parse('2023-09-22')
+$tip.Title = 'Avoid Array addition'
+$tip.TipText = @'
+Array addition is a expensive and inefficient operation and can, most of the times, be replaced by PowerShell explicit loop assignment.
+
+Use a `List<T>` instead in those cases when adding to a collection while looping is required.
+'@
+$tip.Example = @'
+# Array addition:
+$result = @()
+foreach ($i in 0..10) {
+    $result += $i
+}
+
+# Can be easily replaced with explicit assignemnt:
+$result = foreach ($i in 0..10) {
+    $i
+}
+
+# And, when not possible, a List<T> is recommended:
+$result = [System.Collections.Generic.List[int]]::new()
+foreach ($i in 0..10) {
+    $result.Add($i)
+}
+'@
+$tip.Urls = @(
+	'https://learn.microsoft.com/en-us/powershell/scripting/dev-cross-plat/performance/script-authoring-considerations?view=powershell-7.3#array-addition'
+)
+$tip.MinPowerShellVersion = '5.1'
+$tip.Category = [tiPS.TipCategory]::Other # Community, CoreCmdlet, Editor, Module, Syntax, Terminal, or Other.
```

Comment:
I think this tip would fall under the `Syntax` category.

Reply:
@deadlydog I raised #24 proposing adding `Performance` to the tip enum, if you don't think it's a good idea then I can change my commit to `[tiPS.TipCategory]::Syntax` (I personally do not think this tip is syntax related).

Reply:
Yeah, I saw but haven't had a chance to respond there yet. I'll do that in a bit. Thanks!